### PR TITLE
added export and import of followed contacts to and from  CSV files

### PR DESCRIPTION
**Pull request by tobiasd**
This PR adds

* the export of the *followed* accounts as CSV file that is crafted following the Mastodon CSV file.
* the import of these CSV files

Tested and working as expected in the Vagrant VM.

@annando please have a hint for the `q()` in the export function.

### CODE REVIEW

File: src/Module/Settings/UserExport.php [hunk 1]
```diff
@@ -42,6 +43,7 @@ public static function content()
 		$options = [
 			['settings/userexport/account', L10n::t('Export account'), L10n::t('Export your account info and contacts. Use this to make a backup of your account and/or to move it to another server.')],
 			['settings/userexport/backup', L10n::t('Export all'), L10n::t("Export your accout info, contacts and all your items as json. Could be a very big file, and could take a lot of time. Use this to make a full backup of your account \x28photos are not exported\x29")],
+			['settings/userexport/contactcsv', L10n::t('Export Contacts to CSV'), L10n::t("Export the list of the accounts you are following as CSV file. Compatible to e.g. Mastodon.")],
```
Comment (MrPetovan) [inline src/Module/Settings/UserExport.php]: Please rename to `contacts` if we don't have multiple export formats.

Comment (tobiasd) [reply src/Module/Settings/UserExport.php]: That is something I need to investigate today, if e.g. Diaspora* has some import/export mechanism I would like to add that format as well.

Comment (MrPetovan) [reply src/Module/Settings/UserExport.php]: Fair enough.

Comment (tobiasd) [reply src/Module/Settings/UserExport.php]: I've renamed it, as I've not found other platforms exporting the followed accounts.

File: src/Module/Settings/UserExport.php [hunk 2]
```diff
@@ -134,6 +144,22 @@ private static function exportRow(string $query)
 		return $result;
 	}
 
+	/**
+	 * Export a list of the contacts as CSV file as e.g. Mastodon and Pleroma are doing.
+	 *
+	 * @param App $a the app data
+	 **/
+	private static function exportContactsAsCSV(App $a)
```
Comment (nupplaphil) [inline src/Module/Settings/UserExport.php]: We don't need `App $a` here, so you can savely remove the parameter

File: view/templates/settings/settings.tpl [hunk 3]
```diff
@@ -196,6 +196,17 @@
 </div>
 </div>
 
+<h3 class="settings-heading"><a href="javascript:;">{{$importcontact}}</a></h3>
+<div class="settings-content-block">
+<input type="hidden" name="MAX_FILE_SIZE" value="30720" />
```
Comment (nupplaphil) [inline view/templates/settings/settings.tpl]: Maybe a nitpicking part, but wouldn't it be better to have a `const MAX_FILE_SIZE = 30720;` inside `src\Module\Settings\*` and reference it here?

* its better visible when adapting the code
* you've just to define it once and can use it at both places (avoid forgetting one place or another when adapting the value)

Comment (tobiasd) [reply view/templates/settings/settings.tpl]: Possible yes. The admin should most likely the one controling that value via a configuration setting.

File: src/Module/Settings/UserExport.php [hunk 4]
```diff
@@ -64,17 +66,25 @@ public static function rawContent()
...
 				case "account":
+					header("Content-type: application/json");
+					header('Content-Disposition: attachment; filename="' . $user['nickname'] . '.' . $action . '"');
 					self::exportAccount(self::getApp());
 					exit();
 					break;
+				case "contact":
+					header("Content-type: application/csv");
+					header('Content-Disposition: attachment; filename="' . $user['nickname'] . '-contacts.csv'. '"');
+					self::exportContactsAsCSV(self::getApp());
```
Comment (nupplaphil) [inline src/Module/Settings/UserExport.php]: You've to remove the parameter here as well

File: view/templates/settings/settings.tpl (hunk 3 above)
Comment (nupplaphil) [reply view/templates/settings/settings.tpl]: Do you want to add it to this PR or shall I go on merging? :)

Comment (tobiasd) [reply view/templates/settings/settings.tpl]: I've *just* pushed it.

**Comment by AlfredSK** [comment]
There is a problem with settings -> userexport (Export personal data) now.
When I click on the link in the left side menu to go to https://example.com/settings/userexport I get a white page with an error message:
`replaceMacros: Unable to load template 'file:view/templates/settings/userexport.tpl'`

There is also a PHP warning I will report in the pinned issue.

**Pull request merged**